--- /tmp/term.js.beautiful	2012-09-30 23:18:01.000000000 +0900
+++ term.js	2012-09-30 23:28:47.000000000 +0900
@@ -39,14 +39,12 @@
         for (i = 0; i < this.w; i++) da[i] = c;
         this.lines[y] = da;
     }
-    document.writeln('<table border="0" cellspacing="0" cellpadding="0">');
     for (y = 0; y < this.h; y++) {
-        document.writeln('<tr><td class="term" id="tline' + y + '"></td></tr>');
+        $('#terminal').append('<tr><td class="term" id="tline' + y + '"></td></tr>');
     }
-    document.writeln('</table>');
     this.refresh(0, this.h - 1);
-    document.addEventListener("keydown", this.keyDownHandler.bind(this), true);
-    document.addEventListener("keypress", this.keyPressHandler.bind(this), true);
+    $('#terminal').keydown(this.keyDownHandler.bind(this));
+    $('#terminal').keypress(this.keyPressHandler.bind(this));
     ea = this;
     setInterval(function() {
         ea.cursor_timer_cb();
